src/main/java/edu/kit/ipd/crowdcontrol/objectservice/quality/QualityIdentificator.java

```diff
@@ -0,0 +1,221 @@
+        this.experimentOperations = experimentOperations;
+        this.algorithmOperations = algorithmOperations;
+        this.answerAlgorithms = new HashSet<>();
+        this.ratingAlgorithms = new HashSet<>();
+
+        // Reflection isn't used, that's why algorithms need to be added manually
+        //------------------------------------------------------
+        //
+        //           ADD OTHER ALGORITHMS HERE
+        //
+        //------------------------------------------------------
+        answerAlgorithms.add(new AnswerQualityByRatings());
+        ratingAlgorithms.add(new RatingQualityByDistribution());
+
+
+
+        //Load algorithms in db
+        answerAlgorithms.forEach(algorithm -> algorithmOperations.storeAnswerQualityAlgorithm(new AlgorithmAnswerQualityRecord(algorithm.getAlgorithmName(),algorithm.getAlgorithmDescription()),algorithm.getParams()));
+        ratingAlgorithms.forEach(algorithm -> algorithmOperations.storeRatingQualityAlgorithm(new AlgorithmRatingQualityRecord(algorithm.getAlgorithmName(),algorithm.getAlgorithmDescription()),algorithm.getParams()));
+
+        ratingObservable.subscribe(rating -> this.onNext(rating.getData()));
+
+    }
+
+    /**
+     * This method is performed, if the RATINGS_CREATE-observable emits an event.
+     * All ratings and answers of the experiment will be rated. Furthermore the status of the experiment is checked and
+     * if the criteria are met, it will be shut-down
+     * @param rating which has been created an will be processed
+     */
+    private void onNext(Rating rating) {
+
+        Optional<ExperimentRecord> exp = experimentOperations.getExperiment(rating.getExperimentId());
+        if (!exp.isPresent()) {
+            throw new IllegalArgumentException("Error! Can't retrieve the experiment matching to ID:" + rating.getExperimentId());
+        }
+
+        try {
+            ratingIdentifier = getRatingQualityAlgorithm(exp.get().getAlgorithmQualityRating()).get();
+        } catch (NoSuchElementException e1) {
+            log.fatal("Error! Could not find %s-algorithm. Replacing with default RatingQualityByDistribution-algorithm.", exp.get().getAlgorithmQualityRating(), e1);
+            ratingIdentifier = new RatingQualityByDistribution();
+        }
+        try {
+            answerIdentifier = getAnswerQualityAlgorithm(exp.get().getAlgorithmQualityAnswer()).get();
+        } catch (NoSuchElementException e1) {
+            log.fatal("Error! Could not find %s-algorithm. Replacing with default AnswerQualityByRatings-algorithm.", exp.get().getAlgorithmQualityAnswer(), e1);
+            answerIdentifier = new AnswerQualityByRatings();
+        }
+
+
+        algorithmOperations.getAnswerQualityParams(ratingIdentifier.getAlgorithmName(), exp.get().getIdExperiment()).entrySet().forEach(entry ->{
+            if(entry.getKey().field5().toString().equals("RatingQualityThreshold")){    //DOES THIS COMPARISON WORK ?
```

Comment:
no it does, not it returns the name of the field. what you want is `entry.getKey().getData()`